external/akka/src/main/scala/org/apache/spark/streaming/akka/AkkaUtils.scala

```diff
@@ -0,0 +1,148 @@
+   */
+  def actorStream[T: ClassTag](
+      ssc: StreamingContext,
+      actorSystemCreator: () => ActorSystem,
+      props: Props,
+      name: String,
+      storageLevel: StorageLevel = StorageLevel.MEMORY_AND_DISK_SER_2,
+      supervisorStrategy: SupervisorStrategy = ActorSupervisorStrategy.defaultStrategy
+      ): ReceiverInputDStream[T] = ssc.withNamedScope("actor stream") {
+    val cleanF = ssc.sc.clean(actorSystemCreator)
+    ssc.receiverStream(
+      new ActorReceiverSupervisor[T](cleanF, props, name, storageLevel, supervisorStrategy))
+  }
+
+  /**
+   * Create an input stream with any arbitrary user implemented actor receiver.
+   *
+   * @param jssc the StreamingContext instance
+   * @param actorSystemFactory an ActorSystemFactory to create ActorSystem in executors
+   * @param props Props object defining creation of the actor
+   * @param name Name of the actor
+   * @param storageLevel Storage level to use for storing the received objects
+   * @param supervisorStrategy the supervisor strategy
+   *
+   * @note An important point to note:
+   *       Since Actor may exist outside the spark framework, It is thus user's responsibility
+   *       to ensure the type safety, i.e parametrized type of data received and actorStream
+   *       should be same.
+   */
+  def actorStream[T](
+      jssc: JavaStreamingContext,
+      actorSystemFactory: ActorSystemFactory,
+      props: Props,
+      name: String,
+      storageLevel: StorageLevel,
+      supervisorStrategy: SupervisorStrategy
+    ): JavaReceiverInputDStream[T] = {
+    implicit val cm: ClassTag[T] =
+      implicitly[ClassTag[AnyRef]].asInstanceOf[ClassTag[T]]
+    actorStream[T](
+      jssc.ssc, () => actorSystemFactory.create(), props, name, storageLevel, supervisorStrategy)
+  }
+
+  /**
+   * Create an input stream with any arbitrary user implemented actor receiver.
+   *
+   * @param jssc the StreamingContext instance
+   * @param actorSystemFactory an ActorSystemFactory to create ActorSystem in executors
+   * @param props Props object defining creation of the actor
+   * @param name Name of the actor
+   * @param storageLevel Storage level to use for storing the received objects
+   *
+   * @note An important point to note:
+   *       Since Actor may exist outside the spark framework, It is thus user's responsibility
+   *       to ensure the type safety, i.e parametrized type of data received and actorStream
+   *       should be same.
+   */
+  def actorStream[T](
+      jssc: JavaStreamingContext,
+      actorSystemFactory: ActorSystemFactory,
+      props: Props,
+      name: String,
+      storageLevel: StorageLevel
+    ): JavaReceiverInputDStream[T] = {
+    implicit val cm: ClassTag[T] =
+      implicitly[ClassTag[AnyRef]].asInstanceOf[ClassTag[T]]
+    actorStream[T](jssc.ssc, () => actorSystemFactory.create(), props, name, storageLevel)
+  }
+
+  /**
+   * Create an input stream with any arbitrary user implemented actor receiver.
```

Comment:
`with a user-defined actor. See [[ActorReceiver]] for more details.` 

And then in ActorReceiver docs please make sure there is sufficient example code.